src/controllers/midi/midicontroller.cpp

```diff
@@ -160,53 +160,78 @@ void MidiController::destroyOutputHandlers() {
     }
 }
 
-QString formatMidiMessage(const QString& controllerName,
-                          unsigned char status, unsigned char control,
-                          unsigned char value, unsigned char channel,
-                          unsigned char opCode, mixxx::Duration timestamp) {
+QString MidiController::formatMidiMessage(const QString& controllerName,
```

Comment:
No, it isn't implied. That was actually confusing me for a bit before I got help in ##c++-basic on Freenode. I was getting a linker error when I added it to the class as a protected member function but didn't add the class name to the definition.